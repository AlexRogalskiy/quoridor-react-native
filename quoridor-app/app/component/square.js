import React from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

const styles = StyleSheet.create({
  square: {
    position: "absolute",
    width: 30,
    height: 30,
    backgroundColor: "rgb(65, 50, 43)"
  }
});

const Square = ({ position }) => {
  return (
    <View
      style={[
        styles.square,
        {
          top: position.top,
          left: position.left
        }
      ]}
    />
  );
};

Square.prototype = {
  position: PropTypes.shape({
    top: PropTypes.number.isRequired,
    left: PropTypes.number.isRequired
  }).isRequired
};

export default Square;
