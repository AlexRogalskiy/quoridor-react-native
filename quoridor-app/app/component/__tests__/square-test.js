import React from "react";
import renderer from "react-test-renderer";

import { Square } from "../square";

it("renders correctly", () => {
  const tree = renderer
    .create(<Square position="{top: 0, left: 10}" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe("<Pawn />", () => {
  it("has no child", () => {
    const tree = renderer
      .create(<Square position="{top: 0, left: 10}" />)
      .toJSON();
    expect(tree.children.length).toBe(0);
  });
});
