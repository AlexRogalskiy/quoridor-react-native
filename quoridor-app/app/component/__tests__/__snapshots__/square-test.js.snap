// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(65, 50, 43)",
        "height": 30,
        "position": "absolute",
        "width": 30,
      },
      Object {
        "left": undefined,
        "top": undefined,
      },
    ]
  }
/>
`;
